From: wuyifeng <wuyifeng@nd.com.cn>
Date: Thu, 15 Dec 2022 11:41:05 +0800
Subject: Enable bookmark when login

Fix version: 1.2.1
Enable the bookmark functionality when the user signs in to Promethean.
Relate to AP9-11097 and AP9-11086.
Enable the Settings functionality when the user signs in to Promethean.
Relate to AP9-10283.

---
 .../chrome/browser/KeyboardShortcuts.java     | 27 +++++++++++++++
 .../AppMenuPropertiesDelegateImpl.java        | 8 +++++---
 .../browser/omnibox/LocationBarTablet.java    | 10 +++++++---
 3 files changed, 39 insertions(+), 6 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java b/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java
@@ -12,6 +12,7 @@ import android.view.KeyboardShortcutInfo;

 import androidx.annotation.RequiresApi;

+import org.chromium.base.PrometheanManager;
 import org.chromium.chrome.R;
 import org.chromium.chrome.browser.fullscreen.FullscreenManager;
 import org.chromium.chrome.browser.tab.Tab;
@@ -123,6 +124,7 @@ public class KeyboardShortcuts {
      */
     @RequiresApi(Build.VERSION_CODES.N)
     public static List<KeyboardShortcutGroup> createShortcutGroup(Context context) {
+        final boolean isLogin = PrometheanManager.INSTANCE.isLogin();
         final int ctrlShift = KeyEvent.META_CTRL_ON | KeyEvent.META_SHIFT_ON;

         List<KeyboardShortcutGroup> shortcutGroups = new ArrayList<>();
@@ -155,6 +157,10 @@ public class KeyboardShortcuts {
                 KeyEvent.KEYCODE_L, KeyEvent.META_CTRL_ON);
         addShortcut(context, chromeFeatureShortcutGroup, R.string.keyboard_shortcut_address_bar,
                 KeyEvent.KEYCODE_D, KeyEvent.META_ALT_ON);
+        if (isLogin) {
+            addShortcut(context, chromeFeatureShortcutGroup,
+                R.string.keyboard_shortcut_bookmark_manager, KeyEvent.KEYCODE_B, ctrlShift);
+        }
         shortcutGroups.add(chromeFeatureShortcutGroup);

         KeyboardShortcutGroup webpageShortcutGroup = new KeyboardShortcutGroup(
@@ -173,6 +179,10 @@ public class KeyboardShortcuts {
                 KeyEvent.KEYCODE_0, KeyEvent.META_CTRL_ON);
         addShortcut(context, webpageShortcutGroup, R.string.keyboard_shortcut_help_center,
                 KeyEvent.KEYCODE_SLASH, ctrlShift);
+        if (isLogin) {
+            addShortcut(context, webpageShortcutGroup, R.string.keyboard_shortcut_bookmark_page,
+                KeyEvent.KEYCODE_D, KeyEvent.META_CTRL_ON);
+        }
         shortcutGroups.add(webpageShortcutGroup);

         return shortcutGroups;
@@ -268,6 +278,8 @@ public class KeyboardShortcuts {
                 }
             }

+            final boolean isLogin = PrometheanManager.INSTANCE.isLogin();
+
             switch (keyCodeAndMeta) {
                 case CTRL | KeyEvent.KEYCODE_TAB:
                 case CTRL | KeyEvent.KEYCODE_PAGE_DOWN:
@@ -352,6 +364,21 @@ public class KeyboardShortcuts {
                 case KeyEvent.KEYCODE_BUTTON_START:
                     if (currentTab != null && currentTab.canGoForward()) currentTab.goForward();
                     return true;
+                case CTRL | SHIFT | KeyEvent.KEYCODE_B:
+                    if (!isLogin) {
+                        return false;
+                    }
+                    menuOrKeyboardActionController.onMenuOrKeyboardAction(
+                        R.id.all_bookmarks_menu_id, false);
+                    return true;
+                case KeyEvent.KEYCODE_BOOKMARK:
+                case CTRL | KeyEvent.KEYCODE_D:
+                    if (!isLogin) {
+                        return false;
+                    }
+                    menuOrKeyboardActionController.onMenuOrKeyboardAction(
+                        R.id.bookmark_this_page_id, false);
+                    return true;
             }
         }

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
@@ -29,6 +29,7 @@ import androidx.core.graphics.drawable.DrawableCompat;
 import org.chromium.base.CallbackController;
 import org.chromium.base.CommandLine;
 import org.chromium.base.ContextUtils;
+import org.chromium.base.PrometheanManager;
 import org.chromium.base.metrics.RecordHistogram;
 import org.chromium.base.supplier.ObservableSupplier;
 import org.chromium.base.supplier.OneshotSupplier;
@@ -555,9 +556,10 @@ public class AppMenuPropertiesDelegateImpl implements AppMenuPropertiesDelegate
         menu.findItem(R.id.enter_vr_id).setVisible(isCurrentTabNotNull && shouldShowEnterVr());

         // AP9-5181 Hide Bookmark functionality and Settings
-        menu.findItem(R.id.all_bookmarks_menu_id).setVisible(false);
-        menu.findItem(R.id.bookmark_all_tabs_menu_id).setVisible(false);
-        menu.findItem(R.id.preferences_id).setVisible(false);
+        final boolean isLogin = PrometheanManager.INSTANCE.isLogin();
+        menu.findItem(R.id.all_bookmarks_menu_id).setVisible(isLogin);
+        menu.findItem(R.id.bookmark_all_tabs_menu_id).setVisible(isLogin);
+        menu.findItem(R.id.preferences_id).setVisible(isLogin);
         menu.findItem(R.id.view_source_id).setVisible(false);
         menu.findItem(R.id.exit_id).setVisible(false);

diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java
@@ -11,6 +11,7 @@ import android.util.AttributeSet;
 import android.view.MotionEvent;
 import android.view.View;

+import org.chromium.base.PrometheanManager;
 import org.chromium.ui.base.LocalizationUtils;

 /**
@@ -53,12 +54,13 @@ class LocationBarTablet extends LocationBarLayout {
     @Override
     protected void onFinishInflate() {
         super.onFinishInflate();
+        final boolean isLogin = PrometheanManager.INSTANCE.isLogin();

         mLocationBarIcon = findViewById(R.id.location_bar_status_icon);
         mBookmarkButton = findViewById(R.id.bookmark_button);
         mSaveOfflineButton = findViewById(R.id.save_offline_button);

-        mBookmarkButton.setVisibility(View.GONE);
+        mBookmarkButton.setVisibility(isLogin? View.VISIBLE : View.GONE);
         boolean isRtl = mUrlActionContainer.getLayoutDirection() == LAYOUT_DIRECTION_RTL;
         int urlActionContainerPadding =
                 getResources().getDimensionPixelSize(R.dimen.location_bar_url_action_padding);
@@ -233,8 +235,10 @@ class LocationBarTablet extends LocationBarLayout {
     }

     /* package */ void setBookmarkButtonVisibility(boolean showBookmarkButton) {
-        // From AP9-5181, we will hide the bookmark button.
-        // so this method won't do anything.
+        final boolean isLogin = PrometheanManager.INSTANCE.isLogin();
+        if (isLogin) {
+            mBookmarkButton.setVisibility(showBookmarkButton ? View.VISIBLE : View.GONE);
+        }
     }

     /* package */ void setSaveOfflineButtonVisibility(
--
2.34.1